x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentForInitialNode.java

```diff
@@ -0,0 +1,196 @@
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (options.nonOptionArguments().contains("--explicitly-acknowledge-execution") == false) {
+            throw new UserException(ExitCodes.CONFIG, "This command is not intended for end users");
+        }
+        if (env.settings().hasValue(XPackSettings.ENROLLMENT_ENABLED.getKey()) && false ==
+            XPackSettings.ENROLLMENT_ENABLED.get(env.settings())) {
+            throw new UserException(ExitCodes.NOOP, "Enrollment is explicitly disabled.");
+        }
+        final CommandLineHttpClient client = getClient(env);
+        final CreateEnrollmentToken createEnrollmentToken = getCreateEnrollmentToken(env);
+        keyStoreWrapper = getKeyStoreWrapper(env);
+        ReadBootstrapPassword(env, terminal);
+        checkClusterHealthWithRetries(client, terminal, elasticUser, credentialsPassword, 0, 5, false);
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            changeElasticUserPassword(terminal, client);
+        } else {
+            password = bootstrapPassword;
+        }
+        token = createEnrollmentToken.createKibanaEnrollmentToken(elasticUser, password);
+        fingerprint = createEnrollmentToken.getFingerprint();
+
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            terminal.println("'elastic' user password: " + password);
+        }
+        terminal.println("CA fingerprint: " + fingerprint);
+        terminal.println("Kibana enrollment token: " + token);
+        if (options.nonOptionArguments().contains("--docker")) {
+            nodeToken = createEnrollmentToken.createNodeEnrollmentToken(elasticUser, password);
+            terminal.println("Node enrollment token: " + nodeToken);
+        }
+    }
+
+    @Override
+    public void close() {
+        if (keyStoreWrapper != null) {
+            keyStoreWrapper.close();
+        }
+        if (bootstrapPassword != null) {
+            bootstrapPassword.close();
+        }
+        if (credentialsPassword != null) {
+            credentialsPassword.close();
+        }
+    }
+
+    protected void changeElasticUserPassword(Terminal terminal, CommandLineHttpClient client) throws Exception {
+        final URL passwordChangeUrl = changeElasticUserPasswordUrl(client);
+        final HttpResponse response;
+        password = new SecureString(generatePassword(20));
+        try {
+            response = client.execute("POST", passwordChangeUrl, elasticUser, credentialsPassword,
+                () -> {
+                    XContentBuilder xContentBuilder = JsonXContent.contentBuilder();
+                    xContentBuilder.startObject().field("password", password).endObject();
+                    return Strings.toString(xContentBuilder);
+                }, this::responseBuilder);
+            if (response.getHttpStatus() != HttpURLConnection.HTTP_OK) {
+                terminal.errorPrintln("");
```

Comment:
@albertzaharovits would we be capturing stdout/stderr and piping it somewhere when we spawn this process from the startup script? i.e. do we need to handle output ? If not let's remove all these, otherwise we need to be more specific as to what we're printing so that it makes sense to anyone reading it (users are not aware of our implementation details so some context would be helpful  - what is this password change, who invoked it etc )

Reply:
We decided to not print to output in this class in our last sync right?